internal/sql/WebhookEventRepository.go

```diff
@@ -0,0 +1,227 @@
+	SaveWebhookEventData(webhookEventData *WebhookEventData) error
+	GetWebhookParsedEventDataByEventIdAndUniqueId(eventId int, uniqueId string) (*WebhookEventParsedData, error)
+	SaveWebhookParsedEventData(webhookEventParsedData *WebhookEventParsedData) error
+	UpdateWebhookParsedEventData(webhookEventParsedData *WebhookEventParsedData) error
+	GetCiPipelineMaterialWebhookDataMapping(ciPipelineMaterialId int, webhookParsedDataId int) (*CiPipelineMaterialWebhookDataMapping, error)
+	SaveCiPipelineMaterialWebhookDataMapping(ciPipelineMaterialWebhookDataMapping *CiPipelineMaterialWebhookDataMapping) error
+	UpdateCiPipelineMaterialWebhookDataMapping(ciPipelineMaterialWebhookDataMapping *CiPipelineMaterialWebhookDataMapping) error
+	GetCiPipelineMaterialWebhookDataMappingForPipelineMaterial(ciPipelineMaterialId int)  ([]*CiPipelineMaterialWebhookDataMapping, error)
+	GetWebhookEventParsedDataByIds(ids []int, limit int) ([]*WebhookEventParsedData, error)
+	GetWebhookEventParsedDataById(id int) (*WebhookEventParsedData, error)
+	GetWebhookEventConfigByEventId(eventId int) (*GitHostWebhookEvent, error)
+}
+
+type WebhookEventRepositoryImpl struct {
+	dbConnection *pg.DB
+}
+
+func NewWebhookEventRepositoryImpl(dbConnection *pg.DB) *WebhookEventRepositoryImpl {
+	return &WebhookEventRepositoryImpl{dbConnection: dbConnection}
+}
+
+
+func (impl WebhookEventRepositoryImpl) GetAllGitHostWebhookEventByGitHostId(gitHostId int) ([]*GitHostWebhookEvent, error) {
+	var gitHostWebhookEvents []*GitHostWebhookEvent
+	err := impl.dbConnection.Model(&gitHostWebhookEvents).
+		Column("git_host_webhook_event.*").
+		Relation("Selectors", func(q *orm.Query) (query *orm.Query, err error) {
+			return q.Where("is_active IS TRUE"), nil
+		}).
+		Where("git_host_id =? ", gitHostId).
+		Where("is_active = TRUE ").
+		Select()
+	return gitHostWebhookEvents, err
+}
+
+
+func (impl WebhookEventRepositoryImpl) SaveWebhookEventData(webhookEventData *WebhookEventData) error {
+	_, err := impl.dbConnection.Model(webhookEventData).Insert()
+	return err
+}
+
+func (impl WebhookEventRepositoryImpl) GetWebhookParsedEventDataByEventIdAndUniqueId(eventId int, uniqueId string) (*WebhookEventParsedData, error) {
+	var webhookEventParsedData WebhookEventParsedData
+	err := impl.dbConnection.Model(&webhookEventParsedData).Where("event_id =? ", eventId).Where("unique_id =? ", uniqueId).Select()
+	if err != nil {
+		if util.IsErrNoRows(err) {
+			return nil, nil
+		}
+		return nil, err
+	}
+	return &webhookEventParsedData, nil
+}
+
+func (impl WebhookEventRepositoryImpl) SaveWebhookParsedEventData(webhookEventParsedData *WebhookEventParsedData) error {
+	_, err := impl.dbConnection.Model(webhookEventParsedData).Insert()
+	return err
+}
+
+func (impl WebhookEventRepositoryImpl) UpdateWebhookParsedEventData(webhookEventParsedData *WebhookEventParsedData) error {
+	_, err := impl.dbConnection.Model(webhookEventParsedData).WherePK().Update()
```

Comment:
please test this method, there might be some bug in underlying library

Reply:
Tested. working fine